iroh-net/src/key.rs

```diff
@@ -1,2 +1,183 @@
-//! The private and public keys of a node.
-pub mod node;
+//! Cryptographic key handling for `iroh-net`.
+
+mod encryption;
+
+use std::{
+    fmt::{Debug, Display},
+    str::FromStr,
+};
+
+pub use ed25519_dalek::{
+    Signature, SigningKey as SecretKey, VerifyingKey as PublicKey, PUBLIC_KEY_LENGTH,
+};
+use serde::{Deserialize, Serialize};
+use ssh_key::LineEnding;
+
+pub use self::encryption::SharedSecret;
+
+/// A keypair.
+#[derive(Clone, Debug)]
+pub struct Keypair {
```

Comment:
nit, should this be `KeyPair` to be consistent rust naming?

Reply:
maybe, i also don’t like having Keypair, SecretKey and PublicKey, so I might combine this with SecretKey

Reply:
called `SecretKey` now